src/pages/HomePage/pages/LifeCenter/LifeCenter.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useState } from "react";
+// import PageOutline from "../../Components/PageOutline";
+import TabSelection from "@/pages/HomePage/Components/reusable/TabSelection";
+import { Button } from "@/components";
+import Modal from "@/components/Modal";
+import { HeaderControls } from "@/components/HeaderControls";
+import LifeCenterForm from "./components/LifeCenterForm";
+import LifeCenterCard from "./components/LifeCenterCard";
+import useWindowSize from "@/CustomHooks/useWindowSize";
+import { useFetch } from "@/CustomHooks/useFetch";
+import { api } from "@/utils/api/apiCalls";
+import GridComponent from "../../Components/reusable/GridComponent";
+
+export type LifeCenterType = {
+  name: string;
+  description: string;
+  location: string;
+  meeting_dates: string[];
+  num_of_members: number;
+  num_of_souls_won: number;
+  id?: string;
+};
+
+export default function LifeCenter() {
+  const [selectedTab, setSelectedTab] = useState("Life Center");
+  const [showModal, setShowModal] = useState(false);
+  const windowSize = useWindowSize();
+  const { data } = useFetch(api.fetch.fetchAllLifeCenters);
+
+  const handleTabSelect = (tab: string) => {
+    setSelectedTab(tab);
+  };
+
+  const [lifeCenters, setLifcenters] = useState<LifeCenterType[]>([]);
+  useEffect(() => {
+    if (data?.data.length) {
+      setLifcenters([...data.data]);
+    }
+  }, [data]);
+
+  const toggleModal = () => {
+    setShowModal((prev) => !prev);
+  };
+
+  const [initialValues, setInitialValues] = useState<LifeCenterType>({
+    description: "",
+    location: "",
+    meeting_dates: [],
+    name: "",
+    num_of_members: 0,
+    num_of_souls_won: 0,
+    id: "",
+  });
+
+  const addToList = (item: LifeCenterType) => {
+    setLifcenters((prev) => [item, ...prev]);
+    toggleModal();
+  };
+
+  const handleEdit = (lifeCenter: LifeCenterType) => {
+    setInitialValues(lifeCenter);
+    toggleModal();
+  };
+  const handleDelete = (id: string) => {
+    setLifcenters((prev) => prev.filter((item) => item.id !== id));
+  };
+
+  const addLifeCenter = () => {
+    setInitialValues({
+      description: "",
+      location: "",
+      meeting_dates: [],
+      name: "",
+      num_of_members: 0,
+      num_of_souls_won: 0,
+      id: "",
+    });
+    toggleModal();
+  };
+
+  const editItem = (item: LifeCenterType) => {
+    const updateLC = lifeCenters.find((lc) => lc.id === item.id);
+
+    const updateAllLCs = lifeCenters.map((lc) =>
```

Comment:
best practice let  do it as a callback inside the setter removes problem with stale values bla bla bla the docs long I can share it though if you want 
you can change the whole function to 

`const editItem = (item: LifeCenterType) => {
  setLifcenters((prev) =>
    prev.map((lc) => (lc.id === item.id ? item : lc))
  );
  toggleModal();
};`